File: frontend/src/app/model/order.ts
import {User} from "./user";

export interface Order {
  id: number;
  product: string;
  function: OrderFunction;
  userId: number;
  user: User;
}

export enum OrderFunction {
  RECEIVED ='Received',
  ACCEPT = 'Accept',
  REJECT = 'Reject'
}

export const FunctionMapping = {
  [OrderFunction.RECEIVED]: "Received",
  [OrderFunction.ACCEPT]: "Accept",
  [OrderFunction.REJECT]: "Reject"
}

File: api/src/main/java/com/example/testprojectrestaurant/model/Order.java
package com.example.testprojectrestaurant.model;

import lombok.*;
import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;
    @Column(name = "product")
    private String product;
    @Enumerated(EnumType.STRING)
    @Column(name = "order_function")
    private Status status;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    public enum Status {
        RECEIVED("RECEIVED"),
        ACCEPT("ACCEPT"),
        REJECT("REJECT");

        Status(String value) {
        }
    }

}
